java/code/src/com/redhat/rhn/testing/MockServletContext.java

```diff
@@ -0,0 +1,674 @@
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.EventListener;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterRegistration;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.Servlet;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRegistration;
+import javax.servlet.SessionCookieConfig;
+import javax.servlet.SessionTrackingMode;
+import javax.servlet.descriptor.JspConfigDescriptor;
+
+/**
+ * Mock implementation of ServletContext for testing purposes.
+ * This class provides stub implementations of all ServletContext methods.
+ */
+public class MockServletContext implements ServletContext {
+
+    private Map<String, Object> attributes = new HashMap<>();
+    private Set<String> resourcePaths;
+    private List<String> realPaths = new ArrayList<>();
+    private int realPathIndex = 0;
+    private URL resource;
+    private Map<String, String> initParameters = new HashMap<>();
+    private RequestDispatcher requestDispatcher;
+
+    /**
+     * Gets all servlets registered in this context.
+     * @return empty enumeration for mock implementation
+     */
+    public Enumeration<Servlet> getServlets() {
+        return Collections.emptyEnumeration();
+    }
+
+
+    /**
+     * Sets the resource to be returned by getResource().
+     * @param resourceIn the URL resource to set
+     */
+    public void setResource(URL resourceIn) {
+        this.resource = resourceIn;
+    }
+
+    /**
+     * Sets the resource to be returned by getResource().
+     * @param resourceIn the URL resource to set
+     * @deprecated replaced by {@link #setResource(URL)}
+     */
+    @Deprecated
+    public void setupGetResource(URL resourceIn) {
+        this.resource = resourceIn;
+    }
+
+
+
+    /**
+     * Gets a URL to the resource mapped to the specified path.
+     * @param string the path to the resource
+     * @return the resource URL
+     */
+    public URL getResource(String string) {
+        return resource;
+    }
+
+    /**
+     * Sets the resource paths to be returned by getResourcePaths().
+     * @param resourcePathsIn the set of resource paths to set
+     */
+    public void setResourcePaths(Set<String> resourcePathsIn) {
+        this.resourcePaths = resourcePathsIn;
+    }
+
+    /**
+     * Gets a directory-like listing of all the paths to resources within the web application.
+     * @param string the partial path used to match the resources
+     * @return the set of resource paths
+     */
+    public Set<String> getResourcePaths(String string) {
+        return resourcePaths;
+    }
+
+    @Override
+    /**
+     * Gets the context path of the web application.
+     * @return empty string for mock implementation
+     */
```

Comment:
Shouldn't the annotations go below the javadoc comment? Seems we don't have any checkstyle rule enforcing either style ...